FLAnimatedImage/FLAnimatedImageFrameCache.m

```diff
@@ -125,14 +128,21 @@ - (instancetype)initWithFrameCount:(NSUInteger)frameCount
         _cachedFramesForIndexes = [cachedFramesForIndexes copy];
 
         CGFloat animatedImageDataSize = frameSize * (_frameCount - skippedFrameCount) / MEGABYTE;
-        if (animatedImageDataSize <= FLAnimatedImageDataSizeCategoryAll) {
-            _frameCacheSizeOptimal = _frameCount;
-        } else if (animatedImageDataSize <= FLAnimatedImageDataSizeCategoryDefault) {
-            // This value doesn't depend on device memory much because if we're not keeping all frames in memory we will always be decoding 1 frame up ahead per 1 frame that gets played and at this point we might as well just keep a small buffer just large enough to keep from running out of frames.
-            _frameCacheSizeOptimal = FLAnimatedImageFrameCacheSizeDefault;
-        } else {
-            // The predicted size exceeds the limits to build up a cache and we go into low memory mode from the beginning.
-            _frameCacheSizeOptimal = FLAnimatedImageFrameCacheSizeLowMemory;
+        switch (strategy) {
+            case FLAnimatedImagePreferFrameCacheStrategyBest:
+                _frameCacheSizeOptimal = FLAnimatedImageFrameCacheSizeDefault;
```

Comment:
I feel strange that the best use `default` size, but default doesn't.

Reply:
Maybe we can change the naming from `FLAnimatedImageFrameCacheSizeDefault` to `FLAnimatedImageFrameCacheSizeBest` to reduce confusion

Reply:
or somthing OptimziedForDataSize?

Reply:
Like this?

```swift
switch (strategy) {
   case FLAnimatedImagePreferFrameCacheStrategyBest:
      _frameCacheSizeOptimal = FLAnimatedImageFrameCacheSizeBest;
   break;
     case FLAnimatedImagePreferFrameCacheStrategyDataSizeOptimazed:
   ...
```